src/main/kotlin/org/partiql/scribe/targets/spark/SparkAstToSql.kt

```diff
@@ -1,6 +1,167 @@
 package org.partiql.scribe.targets.spark
 
+import org.partiql.ast.Ast.exprPathStepField
+import org.partiql.ast.DataType
+import org.partiql.ast.FromExpr
+import org.partiql.ast.FromType
+import org.partiql.ast.Identifier
+import org.partiql.ast.Literal
+import org.partiql.ast.SelectItem
+import org.partiql.ast.expr.ExprBag
+import org.partiql.ast.expr.ExprLit
+import org.partiql.ast.expr.ExprTrim
+import org.partiql.ast.expr.PathStep
+import org.partiql.ast.expr.TrimSpec
+import org.partiql.ast.sql.SqlBlock
 import org.partiql.scribe.ScribeContext
+import org.partiql.scribe.problems.ScribeProblem
 import org.partiql.scribe.sql.AstToSql
+import org.partiql.scribe.sql.utils.concat
+import org.partiql.scribe.sql.utils.list
 
-public open class SparkAstToSql(context: ScribeContext) : AstToSql(context)
+public open class SparkAstToSql(context: ScribeContext) : AstToSql(context) {
+    private val listener = context.getProblemListener()
+
+    override fun visitFromExpr(
+        node: FromExpr,
+        tail: SqlBlock,
+    ): SqlBlock {
+        var h = tail
+        h =
+            when (node.fromType.code()) {
+                FromType.SCAN -> h
+                else ->
+                    listener.reportAndThrow(
+                        ScribeProblem.simpleError(
+                            ScribeProblem.UNSUPPORTED_AST_TO_TEXT_CONVERSION,
+                            "${node.fromType.name()} is an unsupported feature by spark",
+                        ),
+                    )
+            }
+        h = visitExprWrapped(node.expr, h)
+        h = if (node.asAlias != null) h concat " AS ${node.asAlias!!.sql()}" else h
+        // AT and BY should be stopped by feature validation.
+        return h
+    }
+
+    override fun visitIdentifier(
```

Comment:
This is turning a qualified identifier into, for example, `x.y.z`?

Reply:
More specifically, making sure to quote with Spark-specific backticks, ``` `x`.`y`.`z` ```